src/main/java/de/siegmar/logbackgelf/pool/SimpleObjectPool.java

```diff
@@ -87,11 +89,15 @@ public T borrowObject() throws InterruptedException {
             }
         }
 
-        return needToEvict(pooledObject) ? recycle(pooledObject) : pooledObject;
+        final T result = needToEvict(pooledObject) ? recycle(pooledObject) : pooledObject;
+        result.borrow();
+        return result;
     }
 
     private boolean needToEvict(final T pooledObject) {
-        return maxLifeTime < 0 || pooledObject.lifeTime() > maxLifeTime;
+        boolean beyondMaxLifeTime = maxLifeTime > 0 && pooledObject.lifeTime() > maxLifeTime;
+        boolean beyondMaxLastBorrowedTime = maxLastBorrowedTime >= 0 && (System.currentTimeMillis() - pooledObject.lastBorrowed()) > maxLastBorrowedTime;
```

Comment:
The call to `System.currentTimeMillis()` (and the rest of this line) is unnecessary if beyondMaxLifeTime is already true. Therefor I'd like to simplify this method a bit and use two if-blocks with corresponding returns.